src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java

```diff
@@ -5602,6 +5608,49 @@ private void validateAuditableProperties(Map<String, Object> givenProperties, No
         assertFalse(givenProperties.get("cm:modified").equals(node.getModifiedByUser().getDisplayName()));
     }
 
+    @Test public void testPrimaryPath() throws Exception
+    {
+        setRequestContext(user1);
+        String myNodeId = getMyNodeId();
+        // /Company Home/User Homes/user<timestamp>/folder_A
+        String folderAName = "folder_A";
+        Folder folderA = createFolder(myNodeId, folderAName);
+
+        AuthenticationUtil.setFullyAuthenticatedUser(user1);
+
+        // /Company Home/User Homes/user<timestamp>/folder_A/folder_B
+        String folderBName = "folder_B";
+        Folder folderB = createFolder(folderA.getId(), folderBName);
+        NodeRef folderANodeRef = new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, folderA.getId());
+        Path folderAPath = nodeService.getPath(folderANodeRef);
+        Path.ChildAssocElement pathElement = (Path.ChildAssocElement) folderAPath.last();
+        String qNameLocalName = pathElement.getRef().getQName().getLocalName();
+        assertTrue(folderAName.equals(qNameLocalName));
+
+        // update node
+        // /Company Home/User Homes/user<timestamp>/folder_A01
+        folderA.setName("folder_A01");
+        put(URL_NODES, folderA.getId(), toJsonAsStringNonNull(folderA), null, 200);
+        folderAPath = nodeService.getPath(folderANodeRef);
+        pathElement = (Path.ChildAssocElement) folderAPath.last();
+        qNameLocalName = pathElement.getRef().getQName().getLocalName();
+        assertFalse(folderAName.equals(qNameLocalName));
+
+        folderA.setName("folder_A02");
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("cm:name", "folder_A03");
+        folderA.setProperties(properties);
+        put(URL_NODES, folderA.getId(), toJsonAsStringNonNull(folderA), null, 200);
+
+        folderAPath = nodeService.getPath(folderANodeRef);
```

Comment:
Please check the full path.
Also it would be better to check the path of "folder_B" as the rename should impact children's path as well.

Reply:
I won't do that. I cannot check the full path because it includes the user <timestamp>.

Reply:
user1 is a username with timestamp and it is a variable, see https://github.com/Alfresco/alfresco-remote-api/blob/master/src/test/java/org/alfresco/rest/api/tests/AbstractBaseApiTest.java#L163

Reply:
Done :)